### ISSUE TITLE
AttributeError not detected when same name is used in class attribute

### ISSUE THREAD

**Issue by NCPlayz** [microsoft/pyright#6960]
**Describe the bug**
If you reference the class attribute in the assignment of that class attribute it does not error

**Code or Screenshots**
```python
class Vars:
    def __init__(self) -> None:
        self._values = self._values
        self._keys = self.keys # Cannot access members "keys" for type "Vars"
```

**VS Code extension or command-line**
pyright 1.1.345

**Comment by erictraut** [comment]
This behavior is intended, not a bug.

Instance variables can be assigned from any execution scope (i.e. any method) within a class implementation. Static type checkers cannot generally know which order methods will be called in, so pyright does not attempt to perform any sort of code flow analysis for instance variable assignments. 

The condition you've highlighted in your code sample is admittedly a false negative error, but "fixing" this would produce too many false positive errors.
